### PR TITLE
KryoSerializer without byte buffer

### PR DESCRIPTION
**Pull request by rjhall**
This removes the internal byte buffer used for the Kryo output when serializing.  The reason is to allow the serialization of objects which are so large that they would overflow the buffer.  What's more the logic is simplified.

This does not affect deserialization and preserves the same format for the intermediate files.

### CODE REVIEW

**Comment by jattenberg** [comment]
sick

File: src/jvm/cascading/kryo/KryoSerializer.java [hunk 1]
```diff
@@ -4,65 +4,50 @@
...
+    os.flush();
+  }
+
+  // TODO: Bump the kryo version, add a kryo.reset();
+  public void close() throws IOException {
+    try {
+      if(os != null)
+        os.close();
+    } catch(Exception e) {
+      e.printStackTrace();
```
Comment (avibryant) [inline src/jvm/cascading/kryo/KryoSerializer.java]: Are you intentionally swallowing the exception here instead of re-throwing it? If so, why?

Comment (rjhall) [inline src/jvm/cascading/kryo/KryoSerializer.java]: Good point, this is left over from printf style debugging and should be removed.

Comment (avibryant) [inline src/jvm/cascading/kryo/KryoSerializer.java]: It would have been easier to spot in the diff if you hadn't gratuitously renamed outputStream to os :)

Comment (rjhall) [inline src/jvm/cascading/kryo/KryoSerializer.java]: actually I should fix this and the spacing so the diff isnt so heinous

File: src/jvm/cascading/kryo/KryoSerializer.java [hunk 2]
```diff
@@ -11,58 +12,41 @@
...
     public void serialize(Object o) throws IOException {
-        kryo.writeObject(output, o);
-        byte[] bytes = output.toBytes();
-
-        outputStream.writeInt(bytes.length);
-        outputStream.write(bytes);
-
-        tidyBuffer();
+        // Write output to temorary buffer.
+        ByteArrayOutputStream bho = new ByteArrayOutputStream();
```
Comment (jattenberg) [inline src/jvm/cascading/kryo/KryoSerializer.java:32]: I think it's useful to have longer variable names, and i'd especially avoid single letter variables

File: src/jvm/cascading/kryo/KryoSerializer.java [hunk 3]
```diff
@@ -11,58 +12,41 @@
...
     }
 
     // TODO: Bump the kryo version, add a kryo.reset();
     public void close() throws IOException {
-        kryo = null;
-
         try {
-            if( outputStream != null ) {
+            if(outputStream != null)
                 outputStream.close();
```
Comment (jattenberg) [inline src/jvm/cascading/kryo/KryoSerializer.java:46]: is it convention to have a space after the if? if (some statement) { ... 

I think keeping the braces on single line statements is a good pattern, even if it's not strictly required.

Comment (sritchie) [inline src/jvm/cascading/kryo/KryoSerializer.java:46]: Yup, that's right on the space convention.

**Comment by sritchie** [comment]
Guys, this is great. Can you please redo this pull req against the "develop" branch? I've marked this as the default so pull reqs go there from now on.

**Comment by rjhall** [comment]
done

**Pull request closed without merging**